Competitive Coding/Math/Catalan_Numbers/README.md

```diff
@@ -0,0 +1,17 @@
+The 2 different programs in this folder contain the same implementation!
+The programs basically are a python implementation of the Catalan Numbers.
+They're similar to Fibonacci (very slightly)
+
+The 2 different implementations vary only in terms of time taken!<br>
+Basically we can implement in 3 different ways!<br>
+
+* Recursively - Exponential time complexity<br>
+* Dynamic Programming - O(n^2)<br>
+* Binomial Coefficient - O(n) <br><br>
+
+Each of the above 3 implementations have a descending order of time complexities from (1) to (3) <br><br>
+
+The Binomial implementation has the best time complexity while the Recursive implementation has the worst time complexity<br>
+In this folder we have the Binomial and Recursive implementations of Catalan Numbers.<br><br>
+
+Sources : https://www.geeksforgeeks.org/program-nth-catalan-number/ (You can have a look at this to know more about Catalan number implementations and its theory.)
```

Comment:
some examples needed where catalan number comes in like finding no. of polygon triangulation etc.